cv-data-controller/src/test/java/com/trihydro/cvdatacontroller/controller/CdotUpstreamPathControllerTest.java

```diff
@@ -0,0 +1,296 @@
+package com.trihydro.cvdatacontroller.controller;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.trihydro.cvdatacontroller.controller.CdotUpstreamPathController.DistanceCalculator;
+import com.trihydro.cvdatacontroller.controller.CdotUpstreamPathController.PathDirection;
+import com.trihydro.library.helpers.CdotGisConnector;
+import com.trihydro.library.helpers.Utility;
+import com.trihydro.library.model.Milepost;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+class CdotUpstreamPathControllerTest extends TestBase<CdotUpstreamPathController> {
+  private final String ROUTE_ID = "025A"; // I-25
+  private final String PATH_TO_ROUTE_JSON_TEST_DATA = "src/test/resources/com/trihydro/cvdatacontroller/controller/cdotRouteResponseForI25_First30Mileposts.json";
+
+  @Mock
+  CdotGisConnector cdotGisService;
+
+  @Mock
+  Utility utility;
+
+  @SuppressWarnings("deprecation")
+  List<Milepost> getMockMileposts() throws IOException {
+    String routeJsonString = new String(Files.readAllBytes(Paths.get(PATH_TO_ROUTE_JSON_TEST_DATA)));
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode rootNode = objectMapper.readTree(routeJsonString);
+    JsonNode pathNode = rootNode.path("features").get(0).path("geometry").path("paths").get(0);
+    List<Milepost> mileposts = new ArrayList<Milepost>();
+    for (JsonNode node : pathNode) {
+      Milepost milepost = new Milepost();
+      milepost.setCommonName(ROUTE_ID);
+      BigDecimal latitude = new BigDecimal(node.get(1).asText()).setScale(14, BigDecimal.ROUND_HALF_UP);
+      BigDecimal longitude = new BigDecimal(node.get(0).asText()).setScale(14, BigDecimal.ROUND_HALF_UP);
+      milepost.setLatitude(latitude);
+      milepost.setLongitude(longitude);
+      mileposts.add(milepost);
+    }
+    return mileposts;
+  }
+
+  @BeforeEach
+  void setUp() {
+    uut = new CdotUpstreamPathController();
+    uut.InjectDependencies(cdotGisService);
```

Comment:
suggestion: to allow use of the constructor instead of InjectDependencies
```suggestion
    uut = new CdotUpstreamPathController(cdotGisService);
```

Reply:
Addressed